maint(docs): Updates the CUDA guide (#1626)
canonical/ubuntu-pro-for-wsl: docs/howto/gpu-cuda.md

* Windows 11 (recommended) or Windows 10 with minimum version 21H2 on a physical machine
* NVIDIA graphics card and administrative permission to install device drivers (see also [NVIDIA's system requirements for CUDA](https://docs.nvidia.com/cuda/cuda-installation-guide-microsoft-windows/))

> * Ubuntu 24.04 or higher installed on WSL 2 (different versions of CUDA will work on older versions of Ubuntu, here we'll use the latest of both)

Reviewer: ```suggestion
* Ubuntu 24.04 or higher installed on WSL 2 (different versions of CUDA will work on older versions of Ubuntu, but here we'll use the latest of both)
```

* Broadband internet connection able to download a few GB of data

> ```{tip}

Reviewer: There's a few too many note blocks bunched at the top (not introduced in this PR).

Something to revisit in a later PR I think, once the main updates are made.

Normally, the CUDA toolkit for Linux comes packaged with the device driver for the GPU. On WSL 2, the CUDA driver used is part of the Windows driver installed on the system, and, therefore, care must be taken **not** to install this Linux driver.
```

The following commands will install the WSL-specific CUDA toolkit version 13.2 on Ubuntu 24.04 AMD64 architecture. Be aware that older versions of CUDA (<=10) don’t support WSL 2. Also notice that attempting to install the CUDA toolkit packages straight from the Ubuntu repository (`cuda`, `cuda-13`, or `cuda-drivers`) will attempt to install the Linux NVIDIA graphics driver, which is not what you want on WSL 2.

Navigate to the [CUDA Downloads page](https://developer.nvidia.com/cuda-downloads) and select the appropriate options for your system. In this case, we will select:

* Operating System: Linux
* Architecture: x86_64
* Distribution: WSL-Ubuntu
* Version: 2.0
* Installer Type: deb (network)

> The website renders a series of commands matching the options you selected. The last line installs

Reviewer: ```suggestion
The website shows commands that correspond to your selections. The last line installs
```

a specific version of the CUDA toolkit compatible with WSL 2 distro instances. Make sure to adjust
the version to suit your specific needs.

> For our example, the output will be the following:

Reviewer: ```suggestion
For our example, the commands shown on the website are:
```

Trying to avoid using "output" in the sense of web render and "output" again in the sense of terminal output a few lines below.

```{code-block} text
$ wget https://developer.download.nvidia.com/compute/cuda/repos/wsl-ubuntu/x86_64/cuda-keyring_1.1-1_all.deb

$ sudo dpkg -i cuda-keyring_1.1-1_all.deb

$ sudo apt-get update

$ sudo apt-get -y install cuda-toolkit-13-2
```

> Once complete, you should see a series of outputs that end in `done.`:

Reviewer: ```
Once complete, you should see a series of outputs that end in `done.`:
```

Are we missing an instruction here that tells the user to copy and run the commands, or is it obvious from the browser UI? Even so, we should probably have something like "After running the commands," rather than "Once complete,".

Author: Indeed, and the exact outputs keep changing.

Navigate to the [CUDA Downloads page](https://developer.nvidia.com/cuda-downloads) and select the appropriate options for your system. In this case, we will select:

* Operating System: Linux
* Architecture: x86_64
* Distribution: WSL-Ubuntu
* Version: 2.0
* Installer Type: deb (network)

The website shows commands that correspond to your selections. The last line installs
a specific version of the CUDA toolkit compatible with WSL 2 distro instances. Make sure to adjust
the version to suit your specific needs.

For our example, the commands shown on the website are:

```{code-block} text
$ wget https://developer.download.nvidia.com/compute/cuda/repos/wsl-ubuntu/x86_64/cuda-keyring_1.1-1_all.deb

$ sudo dpkg -i cuda-keyring_1.1-1_all.deb

$ sudo apt-get update

$ sudo apt-get -y install cuda-toolkit-13-2
```

> Copy the commands shown in the website and paste them on a terminal to execute the installation.

Reviewer: ```suggestion
Run the commands shown on the website in your Ubuntu instance to execute the installation.
```

Suggested to simplify slightly (assuming a little of the user) and make it extra obvious it's running in Ubuntu (not just any terminal).

Author: Good point!

```{code-block} text
$ cd ~/Dev

$ git clone https://github.com/nvidia/cuda-samples
```

To build the application, you'll need `cmake`. Install it and then go to the cloned repository directory and run it:

```{code-block} text
$ sudo apt install cmake

$ cd ~/Dev/cuda-samples/Samples/1_Utilities/deviceQuery

$ cmake -S . -B build

$ cmake --build build
```

A successful build will look like the screenshot below.

![Terminal output showing the successful compilation of a CUDA sample application.](assets/gpu-cuda/cmake.png)

Once complete, run the application with:

```{code-block} text
$ ./build/deviceQuery
```

> You should see a similar output to the following detailing the functionality of your CUDA setup (the exact results depend on your hardware setup):

Reviewer: ```suggestion
You should see a similar output to the following, detailing the functionality of your CUDA setup (the exact results depend on your hardware setup):
```

> ```text

Reviewer: ```suggestion
```{code-block} text
:class: no-copy

```
